src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -266,7 +266,7 @@ public uint WritePacket(SNIHandle handle, SNIPacket packet, bool sync)
         private static string GetServerNameWithOutProtocol(string fullServerName, string protocolHeader)
         {
             string serverNameWithOutProtocol = null;
-            if (fullServerName.Length > protocolHeader.Length &&
+            if (fullServerName.Length >= protocolHeader.Length &&
```

Comment:
Without this change, cases where only a protocol is provided look like an invalid provider error rather than a connection string parsing failure. This also allows the underlying parsing to handle empty strings in whatever way they choose.